src/v/cluster/topics_frontend.cc

```diff
@@ -424,6 +424,24 @@ errc topics_frontend::validate_topic_configuration(
         }
     }
 
+    const auto& properties = assignable_config.cfg.properties;
+
+    if (
```

Comment:
should we also validate if local retention is disabled and cloud retention enabled ? i.e. segments will never be evicted from local store

Reply:
You mean if local retention is explicitly disabled? I don't see a reason not to do it.

Reply:
I thought about this a bit more and I'm not sure how to go about it. For example, having size based retention on the local log and time based retention on the remote log seems reasonable to me. I guess we could check that if both size and time based local retention is disabled, then total retention is disabled too.

I would say we revisit this when the "total retention" is actually implemented. We should have a clearer picture of what is required then.